cloud/scope/powervs_cluster.go

```diff
@@ -420,7 +420,7 @@ func (s *PowerVSClusterScope) GetVPCID() *string {
 	if s.IBMPowerVSCluster.Spec.VPC != nil && s.IBMPowerVSCluster.Spec.VPC.ID != nil {
 		return s.IBMPowerVSCluster.Spec.VPC.ID
 	}
-	if s.IBMPowerVSCluster.Status.VPC != nil {
+	if s.IBMPowerVSCluster.Status.VPC != nil && s.IBMPowerVSCluster.Status.VPC.ID != nil {
```

Comment:
also I think since we are returing pointer to string, Its not needed to check for it be nil or not, I strongly think it should be handled in the caller, 

I could see this, It should have been handled to check for id is nil or not. 
```
vpcDetails, _, err := s.IBMVPCClient.GetVPC(&vpcv1.GetVPCOptions{
		ID: s.GetVPCID(),
	})
```

Reply:
Handled it wherever required, PTAL